swarm/swarm.go

```diff
@@ -175,18 +175,24 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 	if err := nodeID.UnmarshalText([]byte(config.NodeID)); err != nil {
 		return nil, err
 	}
+
+	syncing := stream.SyncingAutoSubscribe
+	if !config.SyncEnabled || config.LightNodeEnabled {
+		syncing = stream.SyncingDisabled
+	}
+
+	retrieval := stream.RetrievalEnabled
+	if config.LightNodeEnabled {
+		retrieval = stream.RetrievalClientOnly
+	}
+
 	registryOptions := &stream.RegistryOptions{
 		SkipCheck:       config.DeliverySkipCheck,
-		DoSync:          config.SyncEnabled,
-		DoRetrieve:      true,
-		DoServeRetrieve: true,
+		Syncing:         syncing,
+		Retrieval:       retrieval,
```

Comment:
Previously this code defined the RegistryOptions for the default case and then it was overridden in the LightNode case. Now the different options and modes are calculated before with logic that's not straightforward to follow, so it's not clear how many modes can actually exist.

Reply:
There are here two points of view:
* The functional view about the variables: is syncing enabled or not?
* The node view: is it a light node or not

Both of course are valid equally, and I can't see a procedure which can satisfy both.

I vote to keep it as-is, frankly because I am closer to the former, but I have no strong opinion on having it changed.